app/helpers/time_summary_helper.rb

```diff
@@ -53,4 +53,13 @@ def time_summary_user_options
       [user.full_name_with_function, user.id, selected: user == @user]
     end
   end
+
+  def time_summary_url time_summary
+    time_summary.new_record? ? time_summary_index_path : time_summary_path(time_summary)
+  end
+
+  def time_summary_enabled_edit item, date
+    item.model_name == Activity.model_name &&
+      date >= Time.zone.today.weeks_ago(1)
```

Comment:
Lo pondría distinto, algo como:
```suggestion
    item.kind_of? Activity && date >= 1.week.ago
```